File: src/com/app/test/utils/AppCheck.java
package com.app.test.utils;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 * 检测邮箱
 */
public class AppCheck {


    /**
     * 正则邮箱
     */
    public static boolean checkEmail(String email) {
        String check = "^([a-z0-9A-Z]+[-|\\.]?)+[a-z0-9A-Z]@([a-z0-9A-Z]+(-[a-z0-9A-Z]+)?\\.)+[a-zA-Z]{2,}$";
        Pattern regex = Pattern.compile(check);
        Matcher matcher = regex.matcher(email.trim());
        boolean isMatched = matcher.matches();
        return isMatched;
    }
}

File: app/src/main/java/com/app/test/utils/AppCheck.kt
package com.app.test.utils

import java.util.regex.Pattern

/**
 * 检测邮箱
 */
object AppCheck {
    /**
     * 正则邮箱
     */
    fun checkEmail(email: String): Boolean {
        val check = "^([a-z0-9A-Z]+[-|\\.]?)+[a-z0-9A-Z]@([a-z0-9A-Z]+(-[a-z0-9A-Z]+)?\\.)+[a-zA-Z]{2,}$"
        val regex = Pattern.compile(check)
        val matcher = regex.matcher(email.trim { it <= ' ' })
        return matcher.matches()
    }
}